src/Sentry.NLog/SentryNLogOptions.cs

```diff
@@ -0,0 +1,125 @@
+        /// to <see langword="true" />.
+        /// </summary>
+        public bool MarkNLogNamespaceAsExternal { get; set; } = true;
+
+        /// <summary>
+        /// How many seconds to wait after triggering Logmanager.Shutdown() before just shutting down the
+        /// Sentry sdk.
+        /// </summary>
+        public int ShutdownTimeoutSeconds
+        {
+            get => ShutdownTimeout.Seconds;
+            set => ShutdownTimeout = TimeSpan.FromSeconds(value);
+        }
+
+        /// <summary>
+        /// Minimum log level for events to trigger a send to Sentry. Defaults to <see cref="LogLevel.Error" />.
+        /// </summary>
+        public LogLevel MinimumEventLevel { get; set; } = LogLevel.Error;
+
+        /// <summary>
+        /// Minimum log level to be included in the breadcrumb. Defaults to <see cref="LogLevel.Info" />.
+        /// </summary>
+        public LogLevel MinimumBreadcrumbLevel { get; set; } = LogLevel.Info;
+
+        /// <summary>
+        /// Set this to <see langword="true" /> to ignore log messages that don't contain an exception.
+        /// </summary>
+        public bool IgnoreEventsWithNoException { get; set; } = false;
+
+        /// <summary>
+        /// Determines whether event-level properties will be sent to sentry as additional data. Defaults to <see langword="true" />.
+        /// </summary>
+        /// <seealso cref="SendEventPropertiesAsTags" />
+        public bool SendEventPropertiesAsData { get; set; } = true;
+
+        /// <summary>
+        /// Determines whether event properties will be sent to sentry as Tags or not. Defaults to <see langword="false" />.
+        /// </summary>
+        public bool SendEventPropertiesAsTags { get; set; } = false;
+
+        /// <summary>
+        /// Determines whether or not to include event-level data as data in breadcrumbs for future errors.
+        /// Defaults to <see langword="false" />.
+        /// </summary>
+        public bool IncludeEventDataOnBreadcrumbs { get; set; } = false;
+
+        /// <summary>
+        /// Any additional tags to apply to each logged message.
+        /// </summary>
+        [ArrayParameter(typeof(TargetPropertyWithContext), "tag")]
+        public IList<TargetPropertyWithContext> Tags { get; } = new List<TargetPropertyWithContext>();
+
+        [Advanced]
+        public bool InitializeSdk { get; set; } = true;
+
+        [Advanced]
+        public bool EnableDiagnosticConsoleLogging
+        {
+            get => _enableDiagnosticsLogging;
+
+            set
+            {
+                if (value)
+                {
+                    Debug = true;
+                    DiagnosticLogger = _diagnosticsLogger.Value;
+                    _enableDiagnosticsLogging = true;
+                }
+                else if (_enableDiagnosticsLogging)
+                {
+                    Debug = false;
+                    DiagnosticLogger = null;
+                    EnableDiagnosticConsoleLogging = false;
```

Comment:
This will recurse and there's no code path for it when `value` is false.
Did you intend to set `_enableDiagnosticsLogging = false`?

Reply:
Whoops, yeah. But kind of like the extension overload, I was mainly just using this for debugging the implementation, but it's probably fine to just remove it, since it can be configured manually in the event that it is desired.